lib/active_fulfillment/services/amazon_mws.rb

```diff
@@ -195,27 +196,31 @@ def message_from(response)
 
     def parse_document(xml)
       begin
-        document = REXML::Document.new(xml)
-      rescue REXML::ParseException
-        return { :success => FAILURE }
+        document = Nokogiri::XML(xml)
+      rescue Nokogiri::XML::SyntaxError
+        return XML_FAILURE_RESPONSE
       end
     end
 
+    TRACKING_NUMBER_QUERY = 'FulfillmentShipmentPackage > member > TrackingNumber'.freeze
```

Comment:
Derp, it was a misunderstanding. We can indeed call freeze inside method calls without issues